crypto-ffi/bindings/swift/WireCoreCrypto/WireCoreCrypto/CoreCryptoContext.swift

```diff
@@ -0,0 +1,999 @@
+        try await wrapError {
+            try await context.e2eiRotate(conversationId: conversationId)
+        }
+    }
+    
+    func saveX509Credential(enrollment: E2eiEnrollment, certificateChain: String) async throws -> NewCrlDistributionPoints {
+        try await wrapError {
+            try await context.saveX509Credential(
+                enrollment: enrollment.lower(),
+                certificateChain: certificateChain
+            )
+        }
+    }
+    
+    func getCredentialInUse(groupInfo: Data, credentialType: MlsCredentialType) async throws -> E2eiConversationState {
+        try await wrapError {
+            try await context.getCredentialInUse(
+                groupInfo: groupInfo,
+                credentialType: credentialType.lower()
+            ).lift()
+        }
+    }
+    
+    func getDeviceIdentities(conversationId: ConversationId, deviceIds: [ClientId]) async throws -> [WireIdentity] {
+        try await wrapError {
+            try await context.getDeviceIdentities(
+                conversationId: conversationId,
+                deviceIds: deviceIds
+            ).map { $0.lift() }
+        }
+    }
+    
+    func getUserIdentities(conversationId: ConversationId, userIds: [String]) async throws -> [String : [WireIdentity]] {
+        try await wrapError {
+            try await context.getUserIdentities(
+                conversationId: conversationId,
+                userIds: userIds
+            ).mapValues({ $0.map {$0.lift()} })
+        }
+    }
+    
+    func proteusCryptoboxMigrate(path: String) async throws {
+        try await wrapError {
+            try await context.proteusCryptoboxMigrate(path: path)
+        }
+    }
+    
+    func proteusDecrypt(sessionId: String, ciphertext: Data) async throws -> Data {
+        try await wrapError {
+            try await context.proteusDecrypt(sessionId: sessionId, ciphertext: ciphertext)
+        }
+    }
+    
+    func proteusEncrypt(sessionId: String, plaintext: Data) async throws -> Data {
+        try await wrapError {
+            try await context.proteusEncrypt(sessionId: sessionId, plaintext: plaintext)
+        }
+    }
+    
+    func proteusEncryptBatched(sessions: [String], plaintext: Data) async throws -> [String : Data] {
+        try await wrapError {
+            try await context.proteusEncryptBatched(sessions: sessions, plaintext: plaintext)
+        }
+    }
+    
+    func proteusFingerprint() async throws -> String {
+        try await wrapError {
+            try await context.proteusFingerprint()
+        }
+    }
+    
+    func proteusFingerprintLocal(sessionId: String) async throws -> String {
+        try await wrapError {
+            try await context.proteusFingerprintLocal(sessionId: sessionId)
+        }
+    }
+    
+    func proteusFingerprintPrekeybundle(prekey: Data) throws -> String {
+//        try await wrapError {
+            try context.proteusFingerprintPrekeybundle(prekey: prekey)
+//        }
+    }
```

Comment:
Why is this commented out?

Reply:
Mistake. It has to use a non-async error wrapper.